hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/SQLiteDialect.java

```diff
@@ -457,9 +457,9 @@ public ViolatedConstraintNameExtractor getViolatedConstraintNameExtractor() {
 
 	private static final ViolatedConstraintNameExtractor EXTRACTOR =
 			new TemplatedViolatedConstraintNameExtractor( sqle -> {
-				final int errorCode = JdbcExceptionHelper.extractErrorCode( sqle );
+				final int errorCode = JdbcExceptionHelper.extractErrorCode( sqle ) & 0xFF;
```

Comment:
Should't we check if extended result codes are enabled, and only apply the `& 0xFF` in that case? Would be great if that could be done when initializing the dialect, if possible.

Reply:
`&0xff` works in all cases.

Reply:
Oh right, nice.